cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java

```diff
@@ -620,16 +628,7 @@ private static boolean usesStream(FlowSpecification flowSpec, String stream) {
 
   /* -----------------  helpers to return Json consistently -------------- */
 
-  protected static final DatasetRecord makeDataSetRecord(String name, String classname) {
-    return new DatasetRecord("Dataset", name, name, classname);
-  }
-
-  protected static final StreamRecord makeStreamRecord(String name, StreamSpecification specification) {
-    return new StreamRecord("Stream", name, name, GSON.toJson(specification));
+  protected static StreamRecord makeStreamRecord(String name, StreamSpecification specification) {
```

Comment:
It is strange. There is one place where this is called with specification=null, so I can't use the spec to get the name. I tried not to change existing logic, that's why I kept it this way.